scraper/src/libretexts2zim/zimconfig.py

```diff
@@ -0,0 +1,180 @@
+import argparse
+
+from pydantic import BaseModel
+from zimscraperlib.constants import (  # pyright: ignore[reportMissingTypeStubs]
+    MAXIMUM_DESCRIPTION_METADATA_LENGTH,
+    MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH,
+    RECOMMENDED_MAX_TITLE_LENGTH,
+)
+
+
+class InvalidFormatError(Exception):
+    """Raised when a user supplied template has an invalid parameter."""
+
+    pass
+
+
+class ZimConfig(BaseModel):
+    """Common configuration for building ZIM files."""
+
+    # Displayable name for library courses, e.g. Geosciences
+    library_name: str
+    # File name for the ZIM.
+    file_name_format: str
+    # Name for the ZIM.
+    name_format: str
+    # Human readable title for the ZIM.
+    title_format: str
+    # Publisher for the ZIM.
+    publisher: str
+    # Creator of the content in the ZIM.
+    creator: str
+    # Short description for the ZIM.
+    description_format: str
+    # Long description for the ZIM.
+    long_description_format: str | None
+    # Semicolon delimited list of tags to apply to the ZIM.
+    tags: str
+    # Secondary (background) color of ZIM UI
+    secondary_color: str
+
+    @staticmethod
+    def add_flags(parser: argparse.ArgumentParser, defaults: "ZimConfig"):
+        """Adds flags to the given parser with the given defaults."""
+
+        parser.add_argument(
+            "--library-name",
+            help="Display name for the library courses, e.g. Geosciences",
+            required=True,
+        )
+
+        parser.add_argument(
+            "--creator",
+            help=f"Name of content creator. Default: {defaults.creator!r}",
+            default=defaults.creator,
+        )
+
+        parser.add_argument(
+            "--publisher",
+            help=f"Custom publisher name. Default: {defaults.publisher!r}",
+            default=defaults.publisher,
+        )
+
+        parser.add_argument(
+            "--file-name-format",
+            help="Custom file name format for individual ZIMs. "
+            f"Default: {defaults.file_name_format!r}",
+            default=defaults.file_name_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--name-format",
+            help="Custom name format for individual ZIMs. "
+            f"Default: {defaults.name_format!r}",
+            default=defaults.name_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--title-format",
+            help=f"Custom title format for individual ZIMs. "
+            f"Value will be truncated to {RECOMMENDED_MAX_TITLE_LENGTH} chars. "
```

Comment:
I don't think we have a convention again dynamic metadata. 

What we have clearly stated is that `youtube` scraper shouldn't use dynamic description since it is too often wrong / too long / ...

And we have stated that we do not want to create multiple ZIMs in one recipe because it is hard to maintain and forces the use of dynamic metadata which is known to be wrong on Youtube.

Here (and on DevDocs) the approach is quite different:
- we know that we will have multiple recipes to create and maintain, and all of them will be created at once (where is Youtube we are more "on-the-go" with new recipes every now and then)
- we know that the dynamic metadata we propose as a default is going to be fine in at least 99% of the cases
- we know that configuring mostly the same thing in multiple Zimfarm recipes is only going to cause typos and is going to be hard to maintain

So I feel like it make sense to generate these dynamic metadata as default and to fail early should these not match our constraints (and not truncate which is going to create ugly metadata). This way, content editor will be notified should the dynamic metadata be wrong and they can customize it.

What is true is that we could maybe not advertise the possibility to configure dynamic metadata in the Zimfarm, since either the dynamic default is OK or we should use a plain text value known to be ok.